TestResultSummaryService/parsers/BenchmarkParser.js

```diff
@@ -46,116 +49,78 @@ class BenchmarkParser extends Parser {
-                        }
-
-                    // single metric value
-                    } else {
-                        if ( ( curRegexResult = BenchmarkMetric[curBenchmarkNamedRouted]["metrics"][i].regex.exec( curSearchString ) ) !== null ) {
-                            curMetricValue = [parseFloat(curRegexResult[1])];
-                        }
-                    }
-
-                    curTestData["metrics"].push({name: curMetric, value: curMetricValue})
-                }
-
-                // Parse Java version
-                if ( ( curRegexResult = javaVersionRegex.exec( curItr.value ) ) !== null ) {
-                    curJavaVersion = curRegexResult[1];
-                } else {
-                    curJavaVersion = "benchmarkProduct: " + curBenchmarkProduct;
-                }
-                curTestData["javaVersion"] = curJavaVersion;
-
-                if ( ( curRegexResult = javaBuildDateRegex.exec( curJavaVersion ) ) !== null ) {
-                    curJavaBuildDate = curRegexResult[1];
-                }
-                curTestData["jdkBuildDateUnixTime"] = this.convertBuildDateToUnixTime( curJavaBuildDate );
-            }
-
-            tests.push( {
-                testOutput: curItr.value,
-                testResult: isValid ? "PASSED" : "FAILED",
-                testIndex: testIndex++,
-                benchmarkName: curBenchmarkName,
-                benchmarkVariant: curBenchmarkVariant,
-                benchmarkProduct: curBenchmarkProduct,
-                testData: curTestData
-            } );
+            let curBenchmarkVariant = benchmarkVariantRegex.exec( curItr.value )[1];
+            
+            let curBenchmarkProduct = benchmarkProductRegex.exec( curItr.value )[1];
+             
+            let curBenchmarkNamedRouted = BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant];
+    
+            curTestData["javaVersion"] = javaVersionRegex.exec( curItr.value )[1];
+            
+            curTestData["jdkBuildDateUnixTime"] = javaBuildDateRegex.exec( curTestData["javaVersion"] )[1];
+            
+              
+   
+            // parse metrics
+            curTestData['metrics'] = [];
+            for ( let i = 0; i < BenchmarkMetric[curBenchmarkNamedRouted]['metrics'].length; i++ ) 
+            {
+                 curMetric = BenchmarkMetric[curBenchmarkNamedRouted]['metrics'][i].name;
+
+
+                 if(BenchmarkMetric[curBenchmarkNamedRouted].outerRegex === undefined)
+                 {
+                	   curSearchString = curItr.value;
+                 } else {
+                	   curRegexResult = BenchmarkMetric[curBenchmarkNamedRouted].outerRegex.exec(curItr.value);
+                	   curSearchString = curRegexResult[1];
+                   }
+     
+	               curMetricValue = [];
+	               curRegexResult = curSearchString.split(BenchmarkMetric[curBenchmarkNamedRouted]['metrics'][i].regex);
```

Comment:
Let's put a comment here since it's slightly confusing. We should give an example too. For example, for throughput, which contains only 1 value, we'll have an array of length 2. For startup with 8 values, we'll have an array of length 16.